packages/synthetic-chain/src/cli/run.ts

```diff
@@ -45,77 +46,91 @@ const propagateSlogfile = env => {
   ];
 };
 
-export const runTestImage = (proposal: ProposalInfo) => {
-  const [messageFileName, messageFilePath] = createMessageFile();
+export const runTestImage = ({
+  extraDockerArgs = [],
+  proposal,
+  removeContainerOnExit = true,
+}: {
+  extraDockerArgs?: Array<string>;
+  proposal: ProposalInfo;
+  removeContainerOnExit?: boolean;
+}) => {
+  const { name: messageFilePath, removeCallback: removeTempFileCallback } =
+    createMessageFile(proposal);
+  const messageFileName = basename(messageFilePath);
 
   const containerFilePath = `/root/${messageFileName}`;
-  process.env.MESSAGE_FILE_PATH = messageFilePath;
-
-  executeHostScriptIfPresent(proposal, 'before-test-run.sh');
 
-  console.log(`Running test image for proposal ${proposal.proposalName}`);
-  const { name } = imageNameForProposal(proposal, 'test');
-  spawnSync(
-    'docker',
-    [
-      'run',
-      '--env',
-      `MESSAGE_FILE_PATH=${containerFilePath}`,
-      '--mount',
-      `source=${messageFilePath},target=${containerFilePath},type=bind`,
-      '--network',
-      'host',
-      '--rm',
-      ...propagateSlogfile(process.env),
-      name,
-    ],
-    { stdio: 'inherit' },
-  );
+  try {
+    executeHostScriptIfPresent(
+      {
+        MESSAGE_FILE_PATH: messageFilePath,
+      },
+      proposal,
+      'before-test-run.sh',
+    );
 
-  spawnSync('rm', ['--force', messageFilePath]);
+    console.log(`Running test image for proposal ${proposal.proposalName}`);
+    const { name } = imageNameForProposal(proposal, 'test');
+    spawnSync(
+      'docker',
+      [
+        'run',
+        '--env',
+        `MESSAGE_FILE_PATH=${containerFilePath}`,
+        '--mount',
+        `source=${messageFilePath},target=${containerFilePath},type=bind`,
+        '--network',
+        'host',
+        removeContainerOnExit && '--rm',
+        ...propagateSlogfile(process.env),
+        ...extraDockerArgs,
+        name,
+      ]
+        .filter(Boolean)
+        .map(String),
+      { stdio: 'inherit' },
+    );
 
-  executeHostScriptIfPresent(proposal, 'after-test-run.sh');
+    executeHostScriptIfPresent(
+      {
+        MESSAGE_FILE_PATH: messageFilePath,
+      },
+      proposal,
+      'after-test-run.sh',
+    );
+  } catch (err) {
+    removeTempFileCallback();
+    throw err;
+  }
```

Comment:
Don't we want to remove regardless of outcome?

```suggestion
  } finally {
    removeTempFileCallback();
  }
```